### PR TITLE
feat: remove internal gitlab

### PR DESCRIPTION
**Pull request by olevski**
This removes the omnibus gitlab helm deployment that is part of Renku.

We mentioned in our docs that we would deprecate this in the future. I think we should really get rid of it because the more we keep it around the more likely it is that someone will start running with it and then be stuck with the same migration problems that we had to go through.

The easiest/most simple demo of Renku should be done against gitlab.com or any other existing gitlab deployment. Deploying gitlab is NOT a simple thing. And combining that with rolling out Renku in one step is just too daunting.

We also really do not update/maintain/test the Gitlab helm chart. So we run the risk of someone deploying it and having problems or security issues because the helm chart is not up to date.

/deploy

### CODE REVIEW

**Review by rokroskar — CHANGES_REQUESTED** [review]
Thanks @olevski! It will feel good to not be responsible for this anymore :)

File: CHANGELOG.rst [hunk 1]
```diff
@@ -3,11 +3,26 @@
...
+Gitlab deployment as specified in our `documentation <https://renku.readthedocs.io/en/stable/how-to-guides/admin/gitlab.html#migrate-from-renku-bundled-omnibus-gitlab-to-cloud-native-gitlab-helm-chart>`_. 
+before installing this Renku version. You could also simply install the Renku Gitlab Omnibus
+Helm chart separately from Renku, instead of migrating, but this is strongly discouraged as we will not be 
+maintaining or supporting the chart any longer. Gitlab publishes an official Helm chart and
+that is what should be used for deploying Gitlab with Helm.
 
 Internal Changes
 ~~~~~~~~~~~~~~~~
 
+**Features**
```
[the remark this answers is not in the document]
Comment (olevski) [reply CHANGELOG.rst]: Now I dont mention anything like a feature or fix. Just talk about the removal in the first section of the changelog.

[the remark this answers is not in the document]
Comment (olevski) [reply CHANGELOG.rst]: I really dont know how else to formulate this. Let me know what you want me to write and I will add it.

File: scripts/generate-values/base-renku-values.yaml.template [hunk 2]
```diff
@@ -65,9 +43,9 @@ ingress:
     - hosts:
         - {renku_domain}
       secretName: {namespace}-renku-ch-tls
+amalthea:
+  deployCrd: true
```
[the remark this answers is not in the document]
Comment (olevski) [reply scripts/generate-values/base-renku-values.yaml.template]: Yeah my bad - it was from resolving conflicts or merging other feature branches in this one. The diff makes sense now.

**Comment by rokroskar** [comment]
@olevski can we refresh this and get it merged?

**Comment by rokroskar** [comment]
@olevski I rebased this on 0.60.0 and it seems to be all green (https://github.com/SwissDataScienceCenter/renku/pull/3806) - if you agree, I can force-push my branch here and we can get it merged?

**Review by olevski — COMMENTED** [review]
I cannot approve this because I opened the PR. But I looked through the changes and everything looks good.

**Review by aledegano — COMMENTED** [review]
It all looks good.
For completeness I would also add to the file `helm-chart/values.yaml.changelog.md` the deletion of the corresponding values and re-iterating that the embedded Gitlab is gone.

**Comment by rokroskar** [comment]
@aledegano I added the blurb

**Review by aledegano — APPROVED** [review]

**Review by Panaetius — APPROVED** [review]

**Pull request merged**